src/frontend/src/handler/describe.rs

```diff
@@ -369,6 +372,96 @@ fn explain_node<'a>(node: &StreamNode, verbose: bool) -> Pretty<'a> {
     Pretty::simple_record(one_line_explain, fields, children)
 }
 
+pub async fn handle_describe_fragment(
+    handler_args: HandlerArgs,
+    fragment_id: u32,
+) -> Result<RwPgResponse> {
+    let session = handler_args.session.clone();
+    let meta_client = session.env().meta_client();
+    let distributions = &meta_client.list_fragment_distribution().await?;
+    let fragment_dist = distributions
+        .iter()
+        .find(|d| d.fragment_id == fragment_id)
+        .ok_or_else(|| CatalogError::NotFound("fragment", fragment_id.to_string()))?;
```

Comment:
We'd better do the filtering on meta side, by adding a new API, or adding a parameter to the existing API. Fetching all fragments to get 1 fragment seems too wasteful here.

Reply:
good call i added a new method on meta client and stream manager to get fragment by id